src/NServiceBus.Serverless/ServerlessEndpoint.cs

```diff
@@ -46,7 +41,22 @@ public async Task<ErrorHandleResult> ProcessFailedMessage(ErrorContext errorCont
             return await pipeline.PushFailedMessage(errorContext).ConfigureAwait(false);
         }
 
-        async Task InitializeEndpointIfNecessary(TExecutionContext executionContext, CancellationToken token = default)
+        /// <summary>
+        /// Concurrency safe initialization method that allows to get access to the strongly typed configuration.
```

Comment:
"thread-safe" might be the more widely accepted word here?

Reply:
Pushed. Have a look